### PR TITLE
Update README

### LINKED ISSUE #2: Update README
**Description**

Readme has to be updated before initial release. It should explain details about the component, its functions and instructions about how to use the library in other components.

### PR DESCRIPTION
**Pull request by 51-code**
Fixes #2 .

Adds information about the project and instructions for using it, specifically for the now released `PathProperties` object and default configurations.

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -3,15 +3,18 @@
...
+CNF-01 is a library that provides immutable configuration for Java projects. Immutability is achieved by converting the configuration source into an immutable Map. The resulting configuration Map can't be altered even with modifying the underlying source.
 
 == Features
 
 // List your project's features
+- Provides immutable configurations for:
+. Configuration files / path properties (`PathConfiguration`)
+- Default configurations in case the provided configurations from a source are not found or are otherwise broken (`DefaultConfiguration`)
 
 == Documentation
```
Comment (ronja-ui) [inline README.adoc:19]: This section can be taken out

File: README.adoc [hunk 2]
```diff
@@ -3,15 +3,18 @@
...
 // Add a short description of your project. Tell what your project does and what it's used for.
 
-This is a template repository for Teragrep organization.
+CNF-01 is a library that provides immutable configuration for Java projects. Immutability is achieved by converting the configuration source into an immutable Map. The resulting configuration Map can't be altered even with modifying the underlying source.
 
 == Features
 
 // List your project's features
+- Provides immutable configurations for:
+. Configuration files / path properties (`PathConfiguration`)
```
Comment (ronja-ui) [inline README.adoc:16]: There should be more than one listing item in lists. Add more or combine it with the previous one like this: "Provides immutable configurations for `PathConfiguration`" or something like that.

File: README.adoc [hunk 3]
```diff
@@ -20,16 +23,51 @@ See the official documentation on https://docs.teragrep.com[docs.teragrep.com].
...
+
+DefaultConfiguration defaultConfiguration = new DefaultConfiguration(
+        new PathConfiguration("file/path"),
+        defaults
+);
+Map<String, String> configurationMap = defaultConfiguration.asMap();
+----
+
+=== Configuration objects in your project
+The configuration Map from CNF-01 shouldn't be used directly in regular objects. It should only be passed to objects that are responsible for providing configured versions of other objects. In other words, Factories.
```
Comment (ronja-ui) [inline README.adoc:56]: "...for providing configured versions of other objects. In other words, Factories."

The last sentence is missing out a verb. Please combine it with the previous one or modify it into a proper sentence.

File: README.adoc [hunk 4]
```diff
@@ -20,16 +23,51 @@ See the official documentation on https://docs.teragrep.com[docs.teragrep.com].
...
 
 // If your project has limitations, please list them. Otherwise remove this section.
+All planned configuration sources are not yet supported.
 
-== How to [compile/use/implement]
+== How to use
 
 // add instructions how people can start to use your project
+=== Configuration
+All sources of configuration can be used with the `Configuration` interface. It provides the method `asMap()` which returns the immutable configurations.
```
Comment (ronja-ui) [inline README.adoc:32]: Add a line break between titles and paragraphs like this:

````
== Title

paragraph
````

File: README.adoc [hunk 5]
```diff
@@ -20,16 +23,51 @@ See the official documentation on https://docs.teragrep.com[docs.teragrep.com].
...
+An example with `PathConfiguration`:
+[,java]
+----
+Configuration configuration = new PathConfiguration("file/path");
+Map<String, String> configurationMap = configuration.asMap();
+----
+
+=== Default configuration
+You might want to specify default configurations in case something brakes. For example, the file path in the code block above might not be found. In the following example, notice how the Map with default configurations are given through the `ImmutabilitySupportedMap` object to ensure type safety. `DefaultConfiguration` only takes an `ImmutableMap` (provided in CNF-01) as a parameter for the defaults.
+[,java]
```
Comment (ronja-ui) [inline README.adoc:43]: add a line break between paragraphs and code blocks like this:

````
paragraph

[,java]
----
code
----

paragraph
````

File: README.adoc [hunk 6]
```diff
@@ -20,16 +23,51 @@ See the official documentation on https://docs.teragrep.com[docs.teragrep.com].
...
+=== Configuration objects in your project
+The configuration Map from CNF-01 shouldn't be used directly in regular objects. It should only be passed to objects that are responsible for providing configured versions of other objects. In other words, Factories.
+
+Small example with an `Example` object:
+[,java]
+----
+ExampleFactory exampleFactory = new ExampleFactory(configurationMap);
+Example example = exampleFactory.example();
+----
+Here, the logic for instantiating an `Example` is in the `ExampleFactory` object, which receives the configuration map from CNF-01 as a parameter. This ensures that the main object (`Example`) is as clear as it can be.
```
Comment (ronja-ui) [inline README.adoc:64]: Remove parentheses, they're not necessary

**Review by ronja-ui — APPROVED** [review]
Looks good now!

File: README.adoc [hunk 7]
```diff
@@ -3,33 +3,73 @@
 // 2. Update the issues link in Contributing section in the readme file
 // 3. Update the discussion link in config.yml file in .github/ISSUE_TEMPLATE directory
 
-= repo-template
+= CNF-01: Java Configuration Library
```
Comment (kortemik) [inline README.adoc:6]: Teragrep Configuration Library for Java

**Review by kortemik — CHANGES_REQUESTED** [review]

**Review by kortemik — APPROVED** [review]

**Pull request merged**